test/stubadub/core_test.cljc

```diff
@@ -1,36 +1,100 @@
 (ns stubadub.core-test
-  #?(:clj  (:require [clojure.test :refer [deftest is]]
-                     [stubadub.core :refer [with-stub calls-to]])
-     :cljs (:require [cljs.test :refer-macros [deftest is]]
-                     [stubadub.core :refer [calls-to] :refer-macros [with-stub]])))
+  #?(:clj  (:require [clojure.test :refer [deftest is testing]]
+                     [stubadub.core :as sut])
+     :cljs (:require [cljs.test :refer-macros [deftest is testing]]
+                     [stubadub.core :as sut :include-macros true])))
 
 #?(:cljs (defn spit [& args])) ;; these don't exist when running cljs-tests on node.js
 #?(:cljs (defn slurp [& args]))
 
-(deftest calls-to-returns-a-list-of-the-arguments-passed-to-the-stub
+(deftest stub
+  (testing "a stub is a function"
+    (is (ifn? (sut/stub))))
+
+  (testing "a stub returns nil by default"
+    (let [stub (sut/stub)]
+      (is (nil? (stub)))))
+
+  (testing "a stub can take any number of arguments"
+    (let [stub (sut/stub)]
+      (is (nil? (stub 1)))
+      (is (nil? (stub 1 2)))
+      (is (nil? (stub 1 2 3)))
+      ;; more than 20 args to stub not supported on cljs
+      #?(:clj (is (nil? (apply stub (range 1 100)))))))
+
+  (testing "a return value can be specified"
+    (let [stub (sut/stub :returns "not read from disk")]
+      (is (= (stub) "not read from disk"))))
+
+  (let [stub (sut/stub :return-fn {[1] :a [2 3] :b})]
+    (stub 1))
```

Comment:
This doesn't seem to be testing anything?

Reply:
This is what we call REPL litter :)